tgui/packages/tgui/interfaces/CentComViewer.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author Sovexe (https://github.com/Sovexe)
+ * @license ISC
+ */
+
+import { useBackend } from '../backend';
+import { Button, LabeledList, Section, Stack } from '../components';
+import { Window } from '../layouts';
+
+export const CentComViewer = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { banData, key, filterInactive } = data;
+
+  // Parse the JSON string into an array of objects.
+  const banDataJson = JSON.parse(banData);
+
+
+  // Define a functional component for rendering each ban object.
+  const RenderBans = () => {
+    return banDataJson.map((ban, index) => {
+      // Destructure the properties of the ban object.
+      const { active, bannedOn, expires, jobs, reason, sourceName, type, unbannedBy } = ban;
+      const expired = new Date(expires) < new Date();
+
+      const getBanLength = (bannedOn, expires) => {
+        const ONE_SECOND = 1000;
```

Comment:
probably move this out of the CentComViewer functional component into just the file scope?